### PR TITLE
make type of eventName stronger, from string to SubscribeParametersEvent

### PR DESCRIPTION
**Pull request by thiagowfx**

### CODE REVIEW

**Comment by sadym-chromium** [comment]
This is not 100% correct, as allows sending domain-level event like `browsingContext`, which can be used only for subscription / un-subscription for all the domain events. If it is really wanted, please export `EventNames = BrowsingContext.EventNames | Log.EventNames | ...` and reuse it in subscription: `SubscribeParametersEvent = EventNames | typeof BrowsingContext.AllEvents | ...`.

File: src/bidiMapper/domains/events/EventManager.ts [hunk 1]
```diff
@@ -265,7 +268,7 @@ export class EventManager implements IEventManager {
   #markEventSent(
     eventWrapper: EventWrapper,
     channel: string | null,
-    eventName: string
+    eventName: Session.SubscribeParametersEvent
```
Comment (sadym-chromium) [inline src/bidiMapper/domains/events/EventManager.ts:271]: And I believe here it should be the specific event as well
```suggestion
    eventName: Message.EventNames
```

**Comment by thiagowfx** [comment]
The buffer only works if I add casts, e.g.

```
this.#bufferEvent(eventWrapper, eventName as Message.EventNames);
```

Is this OK?

**Comment by sadym-chromium** [comment]
`registerPromiseEvent`'s argument  should be `Message.EventMessage`, not `Session.SubscribeParametersEvent`

**Pull request merged**